pkg/sqlite/load.go

```diff
@@ -654,37 +683,51 @@ func (s *sqlLoader) RemovePackage(packageName string) error {
 		tx.Rollback()
 	}()
 
+	var errors []error
+	// In the case that the @packageName package does not exist,
+	// log an error but don't exit early to ensure that the proper
+	// tables can still cleaned up properly.
+	err = s.checkIfPackageExists(packageName)
```

Comment:
Since database operations are typically expensive, I wonder if it's worth connecting to the database just to log an error, especially when we can possibly do that by checking the results returned from the other exec statements (i.e the number returned as a result of the delete exec statement later that indicates the number of rows affected)

Reply:
I think it's reasonable to point out the cost implications of performing an additional SQL statement during this method, and whether that logic can be deflated to check the rows affected from the other `delete from ...` statements. That said, if we want to keep this method, I would argue that we could simply cache this query as a prepared statement (or just avoid using the query method when interacting with the underlying database) and now we get relatively optimized performance.

Reply:
Hmm, it looks like `db.Query` _may_ create a new prepared statement, but it's difficult to say confidently that's what's happening here, and it's likely that behavior is driver-specific.

Reply:
Thinking more about this, we _could_ just check the rows affected for a delete call, but then we would likely need to do that for both of the delete calls being made, and in the case that method is refactored or extended in the future, we may need to continue to make that comparison, just in case the various tables are out of sync somehow (unlikely looking at the code, but still possible).

If we're fine with that approach, I can update the implementation to check the rows affected for both of the delete calls, and if the number of rows affected is non-zero, i.e. we can make a reasonable assumption that package name is present, then we can make a `packageExists` bool as true and append an error only when that boolean variable is false at the end of the control flow.

@njhale Any thoughts on how to best tackle this? I think there's value in creating a new method just for checking package existence, but it may be overkill as we have alternatives in place that mirror that same logic without too much additional code.